### PR TITLE
docs: improve readability of documentation

### LINKED ISSUE #167: Enhancements to documentation functionality, readability and usability
### Why should this contribution be adopted?
The samples are to be made functional again. Current problems are:
- use of `docker-compose` in advanced sample 1
- missing of catalog fetch in advanced sample 1
- use of non-working property `maxDuration` in sample 2

Further usability improvements would be:
- use of container name instead of container id in streaming sample 2
- use of the logging webserver as docker container instead of JAR

These enhancements improve the usability and readability of the documentation, making it more accessible to new contributors.

### Could it be achieved with existing functionality? If not, why?
Some of these improvements, such as typo fixes and readability enhancements, are straightforward and cannot be achieved with existing functionalities.

### Would this feature limit platform independence in any way? If so, how and why?
No, these changes will not limit platform independence.

### Is it going to be a self-contained feature, or would it cut across the entire code base?
The changes are self-contained, mostly affecting documentation. They are not expected to impact the broader code base or its stability.

### LINKED ISSUE #172: Management API version changed
# Bug Report

When I was trying out the transfer samples for myself, in the [Transfer negociation step](../blob/main/transfer/transfer-01-negotiation/README.md#1-create-an-asset-on-the-provider-side), I found that the Management API version for assets should be "v3" not "v2", i.e the request should be:

```bash
curl -d @transfer/transfer-01-negotiation/resources/create-asset.json \
  -H 'content-type: application/json' http://localhost:19193/management/v3/assets \
  -s | jq
```

At least that is what worked for me not the v2.

### PR DESCRIPTION
**Pull request by MichaelSteinert**
## What this PR changes/adds

This PR:
- improves the readablity of documentation
- fixes some typos
- uses the logging webserver as docker container instead of JAR

- makes use of docker container name instead of container id  in streaming sample 2
- removes non-working property `maxDuration` in streaming sample 2

- makes use of docker compose instead fo docker-compose in advanced sample 1
- adds missing catalog fetch in advanced sample 1

Closes #172 
Closes #167

### CODE REVIEW

File: transfer/streaming/streaming-01-http-to-http/README.md [hunk 1]
```diff
@@ -102,11 +111,11 @@ curl -H 'Content-Type: application/json' -d @transfer/streaming/streaming-01-htt
...
 
 ### Start the transfer
-First we need to set up the receiver server on the consumer side that will receive a call for every message. For this
+First we need to set up the logging webserver on the consumer side that will receive a call for every message. For this
 you'll need to open another terminal shell and run:
 ```shell
-./gradlew util:http-request-logger:build
-HTTP_SERVER_PORT=4000 java -jar util/http-request-logger/build/libs/http-request-logger.jar
+docker build -t http-request-logger util/http-request-logger
+docker run -p 4000:4000 http-request-logger
```
Comment (ndr-brt) [inline transfer/streaming/streaming-01-http-to-http/README.md:118]: Not sure why running with docker should be better than running the jar directly

**Review by ndr-brt — CHANGES_REQUESTED** [review]
Please create an issue as described in the [contributing guide](https://github.com/eclipse-edc/docs/blob/main/CONTRIBUTING.md)

File: transfer/streaming/streaming-01-http-to-http/README.md (hunk 1 above)
Comment (MichaelSteinert) [reply transfer/streaming/streaming-01-http-to-http/README.md:118]: First, I could not run this JAR after building it. Unfortunately, I could not fix this error:
```
Error: Could not find or load main class org.eclipse.edc.samples.util.HttpRequestLoggerServer
Caused by: java.lang.ClassNotFoundException: org.eclipse.edc.samples.util.HttpRequestLoggerServer
```

Secondly, there is a Dockerfile that is already used in other samples

**Comment by MichaelSteinert** [comment]
Corresponding PullRequest #167

File: transfer/streaming/streaming-01-http-to-http/README.md (hunk 1 above)
Comment (ndr-brt) [reply transfer/streaming/streaming-01-http-to-http/README.md:118]: looks like the jar task in the build.gradle.kts misses to include `.class` files in the jar:
```java
from(sourceSets["main"].output)
```

Comment (MichaelSteinert) [reply transfer/streaming/streaming-01-http-to-http/README.md]: Thanks for your help! Now running the JAR works. I have changed the use of container back to JAR.

**Comment by ndr-brt** [comment]
@MichaelSteinert could you take care of make the e2e tests pass? so we can merge then

**Comment by MichaelSteinert** [comment]
@ndr-brt yes. i fixed the e2e tests

**Review by ndr-brt — APPROVED** [review]
thanks!

**Pull request merged**